### DISCUSSION TITLE
Pac模式无法科学上网，只有全局模式可以。

### DISCUSSION THREAD

**Discussion by chloe7sun** [yanue/V2rayU#1360] [Q&A]
刚开始装的3.8.0版本，一切正常；后来更新了4.1.0（在应用程序中直接删除旧软件，安装新的），就出现pac模式无法上网的问题。
重新安装3.8.0，这个问题依然在。
目前就是无论安装哪个版本，pac模式完全不能用，只能全局。
求助！

**Comment by yanue** [comment]
更改端口试下

**Comment by Rounzhou** [comment]
我一开始也是，但是开全局后切换回来过一回就正常了，版本4.1

**Comment by Owen0911** [comment]
我也是同样的问题，开了全局之后，某些软件才能连上网。pac模式开了就不行

**Comment by zhangxiaoyu2046** [comment]
我也有这种情况，v4.1.0, 只有全局可以，PAC不行，求教如何解决？

**Comment by jeffery9** [comment]
是 地址的问题。。  
<img width="520" height="90" alt="image" src="https://github.com/user-attachments/assets/844138b7-d08b-4e67-a8b7-69a19a504625" />
地址要改成 127.0.0.1 才行

**Reply by huacha2** [reply]
感谢，已经解决了

**Comment by jeffery9** [comment]
应该是 代码  里面  https://github.com/yanue/V2rayU/blame/cc256d5f9e818a4d6fc0e72dbd0eeed046fe5d96/V2rayU/Preference/PreferencePac.swift#L181    这个地方 ，其实只需要 使用 本地回环 地址  127.0.0.1

**Comment by jeffery9** [comment]
你反馈：“macOS 15.5 无法使用本机的局域网 IP 来访问 proxy.js，改成 127.0.0.1 地址就可以了。”  
这是一个常见的 macOS 网络安全策略变化带来的问题。

### 原因分析
macOS 从 14.x 开始收紧了本地网络访问权限，许多 API 或服务只允许通过 `127.0.0.1`（环回地址）访问本机服务，而拒绝用本机的局域网 IP（如 `192.168.x.x` 或 `10.x.x.x`）进行访问。这是为了防止本地服务暴露到局域网，提升安全性。

### 你的场景
你访问 `proxy.js` 时使用了本地局域网 IP，导致无法访问。换成 `127.0.0.1` 就能正常访问，说明服务只监听在环回接口或 macOS 拦截了局域网 IP 的 loopback。

### 解决建议
1. **优先使用 127.0.0.1**  
   在配置代理、PAC 或服务监听地址时，建议统一使用 `127.0.0.1`，避免兼容性问题。

2. **修改监听地址（如有必要）**  
   如果你的应用必须让局域网其他设备访问本机服务，需要明确在服务端监听 `0.0.0.0` 或本机实际 IP，但注意 macOS 可能还会有额外权限限制（如防火墙或 App Sandbox）。

3. **代码改进建议**  
   针对 V2rayU 相关代码（如 PreferencePac.swift），建议：
   - 默认用 `127.0.0.1`，不要自动选取本机的局域网 IP。
   - 可以让用户手动设置监听地址，如果有特殊需求。

   例如（伪代码）：
   ```swift
   // 推荐改法
   let defaultSocksAddress = "127.0.0.1"
   socks5Address = defaultSocksAddress
   // 如需支持局域网可选项，再做选择
   ```

### 总结
- macOS 15.x 更推荐使用 127.0.0.1 访问本机服务。
- 局域网 IP 访问在新系统可能不被允许或需要特殊配置。
- 代码和配置建议统一用环回地址，减少兼容性问题。

如果需要具体如何修改 V2rayU 的相关 Swift 代码或代理配置，可以告诉我你的细节需求！

**Comment by tujietg** [comment]
我这边的原因是Pac文件丢失，从Pac设置进入，重新更新一下文件就可以了。
